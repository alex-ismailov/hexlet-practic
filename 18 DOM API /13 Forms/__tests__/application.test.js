import fs from 'fs';
import path from 'path';
import { html } from 'js-beautify';

import run from '../src/application.js';

const htmlOptions = {
  preserve_newlines: false,
  unformatted: [],
};

const fixuturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => html(document.body.innerHTML, htmlOptions);

beforeAll(() => {
  const initHtml = fs.readFileSync(path.join(fixuturesPath, 'index.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run();
});

test('application', () => {
  const form = document.querySelector('.feedback-form');
  expect(getTree()).toMatchSnapshot();

  const { email, name, comment } = form.elements;
  email.value = 'test@email.com';
  name.value = 'Matz';
  comment.value = 'Ruby ze best';
  expect(getTree()).toMatchSnapshot();

  form.dispatchEvent(new Event('submit'));
  expect(getTree()).toMatchSnapshot();
});
