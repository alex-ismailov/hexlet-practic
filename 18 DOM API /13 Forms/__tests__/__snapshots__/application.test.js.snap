// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">
    <h1>Feedback</h1>
    <form class=\\"feedback-form\\">
        <div class=\\"form-group row\\">
            <label for=\\"inputEmail\\" class=\\"col-sm-2 col-form-label\\">Email</label>
            <div class=\\"col-sm-10\\">
                <input type=\\"email\\" class=\\"form-control\\" name=\\"email\\" id=\\"inputEmail\\">
            </div>
        </div>
        <div class=\\"form-group row\\">
            <label for=\\"inputName\\" class=\\"col-sm-2 col-form-label\\">Name</label>
            <div class=\\"col-sm-10\\">
                <input type=\\"text\\" name=\\"name\\" class=\\"form-control\\" id=\\"inputName\\">
            </div>
        </div>
        <div class=\\"form-group row\\">
            <label for=\\"inputComment\\" class=\\"col-sm-2 col-form-label\\">Comment</label>
            <div class=\\"col-sm-10\\">
                <textarea class=\\"form-control\\" name=\\"comment\\" id=\\"inputComment\\"></textarea>
            </div>
        </div>
        <div class=\\"form-group row\\">
            <div class=\\"col-sm-10\\">
                <button type=\\"submit\\" class=\\"btn btn-primary\\">Send</button>
            </div>
        </div>
    </form>
</div>"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">
    <h1>Feedback</h1>
    <form class=\\"feedback-form\\">
        <div class=\\"form-group row\\">
            <label for=\\"inputEmail\\" class=\\"col-sm-2 col-form-label\\">Email</label>
            <div class=\\"col-sm-10\\">
                <input type=\\"email\\" class=\\"form-control\\" name=\\"email\\" id=\\"inputEmail\\">
            </div>
        </div>
        <div class=\\"form-group row\\">
            <label for=\\"inputName\\" class=\\"col-sm-2 col-form-label\\">Name</label>
            <div class=\\"col-sm-10\\">
                <input type=\\"text\\" name=\\"name\\" class=\\"form-control\\" id=\\"inputName\\">
            </div>
        </div>
        <div class=\\"form-group row\\">
            <label for=\\"inputComment\\" class=\\"col-sm-2 col-form-label\\">Comment</label>
            <div class=\\"col-sm-10\\">
                <textarea class=\\"form-control\\" name=\\"comment\\" id=\\"inputComment\\"></textarea>
            </div>
        </div>
        <div class=\\"form-group row\\">
            <div class=\\"col-sm-10\\">
                <button type=\\"submit\\" class=\\"btn btn-primary\\">Send</button>
            </div>
        </div>
    </form>
</div>"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">
    <h1>Feedback</h1>
    <div>
        <p>Feedback has been sent</p>
        <div>Email: test@email.com</div>
        <div>Name: Matz</div>
        <div>Comment: Ruby ze best</div>
    </div>
</div>"
`;
